pkg/webhooks/customresourcedefinitions/customresourcedefinitions.go

```diff
@@ -0,0 +1,202 @@
+	return &customresourcedefinitionsruleWebhook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *customresourcedefinitionsruleWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	//Implement authorized next
+	return s.authorized(request)
+}
+
+func (s *customresourcedefinitionsruleWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	np, err := s.renderCustomResourceDefinition(request)
+	if err != nil {
+		log.Error(err, "Could not render a CustomResourceDefinition from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	if utils.IsProtectedByLabel(np.GetLabels()) || utils.IsProtectedByResourceName(np.GetName()) {
+		log.Info(fmt.Sprintf("%s operation detected on protected CustomResourceDefinition: %s", request.Operation, np.Name))
+		if isAllowedUser(request) {
+			ret = admissionctl.Allowed(fmt.Sprintf("User '%s' in group(s) '%s' can operate on CustomResourceDefinitions", request.UserInfo.Username, strings.Join(request.UserInfo.Groups, ", ")))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+		for _, group := range request.UserInfo.Groups {
+			if privilegedServiceAccountGroupsRe.Match([]byte(group)) {
+				ret = admissionctl.Allowed(fmt.Sprintf("Privileged service accounts in group(s) '%s' can operate on CustomResourceDefinitions", strings.Join(request.UserInfo.Groups, ", ")))
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+		}
+
+		ret = admissionctl.Denied(fmt.Sprintf("User '%s' prevented from accessing Red Mat managed resources. This is in an effort to prevent harmful actions that may cause unintended consequences or affect the stability of the cluster. If you have any questions about this, please reach out to Red Hat support at https://access.redhat.com/support", request.UserInfo.Username))
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+
+	log.Info("Allowing access", "request", request.AdmissionRequest)
+	ret = admissionctl.Allowed("Non managed CustomResourceDefinition")
+	ret.UID = request.AdmissionRequest.UID
+	return ret
+}
+
+// isAllowedUser checks if the user or group is allowed to perform the action
+func isAllowedUser(request admissionctl.Request) bool {
+	if slices.Contains(allowedUsers, request.UserInfo.Username) {
+		return true
+	}
+
+	for _, group := range sreAdminGroups {
+		if slices.Contains(request.UserInfo.Groups, group) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s *customresourcedefinitionsruleWebhook) renderCustomResourceDefinition(req admissionctl.Request) (*customResourceDefinition, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	customResourceDefinition := &customResourceDefinition{}
+
+	if len(req.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(req.OldObject, customResourceDefinition)
+	} else {
+		err = decoder.Decode(req, customResourceDefinition)
+	}
```

Comment:
```suggestion
  err = decoder.Decode(req, customResourceDefinition)
```
Why do we prefer to decode OldObject instead of Object?

Reply:
hm, I tried using the Object, however then normal user and backplane admin could not perform delete method. FWIW this logic is used in other validators
```
pkg/webhooks/namespace/namespace.go-134-        if len(req.OldObject.Raw) > 0 {
pkg/webhooks/namespace/namespace.go-160-        if len(req.OldObject.Raw) == 0 {
pkg/webhooks/pod/pod.go-102-    if len(req.OldObject.Raw) > 0 {
pkg/webhooks/prometheusrule/prometheusrule.go-167-      if len(req.OldObject.Raw) > 0 {
pkg/webhooks/networkpolicies/networkpolicies.go-131-    if len(req.OldObject.Raw) > 0 {
pkg/webhooks/customresourcedefinitions/customresourcedefinitions.go-130-        if len(req.OldObject.Raw) > 0 {
```